jstest/testrunner/devices/connections/sshcom.py

```diff
@@ -60,32 +64,42 @@ def exec_command(self, cmd):
         '''
         Send command over the serial port.
         '''
-        try:
-            if self._no_exec_command:
-                self.send(cmd)
-                response = self.receive()
-
-            else:
-                _, stdout, _ = self.ssh.exec_command(cmd, timeout=self.timeout)
-
-                response = stdout.readline()
-
-        except socket.timeout:
-            raise TimeoutException
-
-        return response
+        self.send(cmd)
+        return self.receive()
 
     def send(self, cmd):
         '''
         Send data over the ssh channel.
         '''
         self.chan.send(cmd + '\n')
 
+    def receive_until_prompt(self):
+        '''
+        Receive data from the server until we get the prompt.
+        '''
+        temp = ''
+        lines = []
+        while self.prompt not in temp:
+            temp += self.chan.recv(1)
+            if temp.endswith('\n'):
+                lines.append(temp)
+                temp = ''
+
+        if lines:
+            if temp != self.prompt:
+                raise InvalidPrompt
+            ret = lines.pop()
+            if 'Stack usage' in ret:
+                # If we run with memstat, the last element will be 'Stack usage', so
+                # we need the element before that.
+                ret = lines.pop()
+            return ret
```

Comment:
You can use the `strip` method of Python to remove the prompt from the received data:

```
return tmp.strip(self.prompt).split('\n')
```